File: exercicios/vjudge/lista-1/b/b.py
def o3_solution(K: int, S: int) -> int:
    count = 0
    for x in range(K + 1):
        for y in range(K + 1):
            for z in range(K + 1):
                if (x + y + z == S):
                    count += 1
    return count

def o2_solution(K: int, S: int) -> int:
    count = 0
    for x in range(K + 1):
        for y in range(K + 1):
            if (S - (x + y) <= K and S - (x + y) >= 0):
                count += 1
    return count


K, S = input().split()
# print(o3_solution(int(K), int(S)))
print(o2_solution(int(K), int(S)))

File: exercicios/vjudge/lista-1/b/b.cpp
#include <bits/stdc++.h>

using namespace std;

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(NULL);

  int K, S;
  cin >> K >> S;

  int count = 0;
  for (int i = 0; i <= K; i++) {
    for (int j = 0; j <= K; j++) {
      // range de z
      if ((S - (i + j)) <= K && (S - (i + j)) >= 0)
        count++;
    }
  }

  cout << count << '\n';

  return 0;
}